Harden expand_path for Windows and safe_log_file_name

Per @nexeh:
```
I hit the root of your next problem now tho in the follow code in core.py...
on line 274 it makes the following call

return_code = _run_process_and_wait(commands, project.expand_path("$dir_dist"), output_file)

which results in the following error:
project.expand_path($dir_dist) c:\Users\jcorson\pythonWorkspace\projectNamehere\target/dist/projectNameHere-0.0.2
BUILD FAILED - [Error 267] The directory name is invalid

You can see the slashs are in the wrong direction for windows. This is becuase in expand path it splits on the windows path sep but its to late for that

def expand_path(self, format_string, *additional_path_elements): elements = [self.basedir] elements += self.expand(format_string).split(PATH_SEPARATOR) elements += list(additional_path_elements) return os.path.join(*elements) 
```

--- a/src/main/python/pybuilder/core.py
+++ b/src/main/python/pybuilder/core.py
@@ -548,7 +548,7 @@ class Project(object):
         elements = [self.basedir]
         elements += self.expand(format_string).split(PATH_SEPARATOR)
         elements += list(additional_path_elements)
-        return os.path.join(*elements)
+        return os.path.normcase(os.path.join(*elements))
 
     def get_property(self, key, default_value=None):
         return self.properties.get(key, default_value)

--- a/src/main/python/pybuilder/utils.py
+++ b/src/main/python/pybuilder/utils.py
@@ -609,4 +609,6 @@ def get_dist_version_string(project, format=" (%s)"):
 
 
 def safe_log_file_name(file_name):
-    return re.sub(r'\<|\>|=|\:|\*|\?|\|\"', '_', file_name)
+    # per https://support.microsoft.com/en-us/kb/177506
+    # per https://msdn.microsoft.com/en-us/library/aa365247
+    return re.sub(r'\\|/|:|\*|\?|\"|<|>|\|', '_', file_name)

